contrib/scrap-ec2-sizes.py

```diff
@@ -0,0 +1,271 @@
+    },
+    'EU (Frankfurt)': {
+        'id': 'eu-central-1',
+        'endpoint': 'ec2.eu-central-1.amazonaws.com',
+        'api_name': 'ec2_eu_central',
+        'country': 'Frankfurt',
+        'signature_version': '4',
+    },
+    # Asia
+    'Asia Pacific (Mumbai)': {
+        'id': 'ap-south-1',
+        'endpoint': 'ec2.ap-south-1.amazonaws.com',
+        'api_name': 'ec2_ap_south_1',
+        'country': 'India',
+        'signature_version': '4',
+    },
+    'Asia Pacific (Singapore)': {
+        'id': 'ap-southeast-1',
+        'endpoint': 'ec2.ap-southeast-1.amazonaws.com',
+        'api_name': 'ec2_ap_southeast',
+        'country': 'Singapore',
+        'signature_version': '2',
+    },
+    'Asia Pacific (Sydney)': {
+        'id': 'ap-southeast-2',
+        'endpoint': 'ec2.ap-southeast-2.amazonaws.com',
+        'api_name': 'ec2_ap_southeast_2',
+        'country': 'Australia',
+        'signature_version': '2',
+    },
+    'Asia Pacific (Tokyo)': {
+        'id': 'ap-northeast-1',
+        'endpoint': 'ec2.ap-northeast-1.amazonaws.com',
+        'api_name': 'ec2_ap_northeast',
+        'country': 'Japan',
+        'signature_version': '2',
+    },
+    'Asia Pacific (Seoul)': {
+        'id': 'ap-northeast-2',
+        'endpoint': 'ec2.ap-northeast-2.amazonaws.com',
+        'api_name': 'ec2_ap_northeast',
+        'country': 'South Korea',
+        'signature_version': '4',
+    },
+    'Asia Pacific (Osaka-Local)': {
+        'id': 'ap-northeast-3',
+        'endpoint': 'ec2.ap-northeast-3.amazonaws.com',
+        'api_name': 'ec2_ap_northeast',
+        'country': 'Japan',
+        'signature_version': '4',
+    },
+    # Not in JSON
+    'China (Beijing)': {
+        'id': 'cn-north-1',
+        'endpoint': 'ec2.cn-north-1.amazonaws.com.cn',
+        'api_name': 'ec2_cn_north',
+        'country': 'China',
+        'signature_version': '4',
+    },
+    'China (Ningxia)': {
+        'id': 'cn-northwest-1',
+        'endpoint': 'ec2.cn-northwest-1.amazonaws.com.cn',
+        'api_name': 'ec2_cn_northwest',
+        'country': 'China',
+        'signature_version': '4',
+    },
+}
+
+
+def download_json():
+    response = requests.get(URL, stream=True)
+    try:
+        return open(FILEPATH, 'r')
+    except IOError:
+        with open(FILEPATH, 'wb') as fo:
+            for chunk in response.iter_content(chunk_size=2**20):
+                if chunk:
+                    fo.write(chunk)
+    return open(FILEPATH, 'r')
+
+
+def get_json():
+    try:
+        return open(FILEPATH, 'r')
+    except IOError:
+        return download_json()
+
+
+def parse():
+    # Set vars
+    sizes = {}
+    regions = {r['id']: r for r in REGION_DETAILS.values()}
+    for region_id in regions:
+        regions[region_id]['instance_types'] = []
+    # Parse
+    json_file = get_json()
+    products_data = ijson.items(json_file, 'products')
+    products_data = next(products_data)
+    for sku in products_data:
```

Comment:
Can you make sure that the output is deterministic since we're committing the result to git? It looks like using `sorted(products_data)` is enough.

Reply:
It's a bit difficult to sort at this level. I use ijson because it's a >250MB JSON.
So to use `sorted` here will try to parse and put that big block in RAM.

Reply:
Running mprof run on the scrap-ec2-sizes, I don't really see a difference between the default mode:

![default](https://user-images.githubusercontent.com/42327/41253204-e40b2de4-6dbf-11e8-88b5-f324a1f89d38.png)

and the sorted() mode:

![sorted](https://user-images.githubusercontent.com/42327/41253208-e7d4f0d6-6dbf-11e8-9b10-e9a49cfbaf0c.png)

And 200MB is acceptable for a script that gets run once in a while, since having a deterministic output is very important to have useable diffs.